### ISSUE TITLE
import error with deepcell.utils.compute_overlap

### ISSUE THREAD

**Issue by ngreenwald** [vanvalenlab/deepcell-tf#265]
After pulling down the latest changes from master, I'm getting an import error when importing deepcell.

I only see a compute_overlap.pyx file, not a compute_overlay.py

`ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-1-cd651fe6dd67> in <module>
      4 import numpy as np
      5 
----> 6 import deepcell

/usr/local/lib/python3.6/dist-packages/deepcell/__init__.py in <module>
     42 from deepcell import training
     43 from deepcell import utils
---> 44 from deepcell import metrics
     45 
     46 from deepcell.layers import *

/usr/local/lib/python3.6/dist-packages/deepcell/metrics.py in <module>
     63 from tensorflow.python.platform import tf_logging as logging
     64 
---> 65 from deepcell.utils.compute_overlap import compute_overlap
     66 
     67 

ModuleNotFoundError: No module named 'deepcell.utils.compute_overlap'`

**Comment by willgraf** [comment]
To use `compute_overlap`, the C extensions must be built using `python setup.py build_ext --inplace`.

I suspect that somehow with the code updates the installed extention got replaced.

**Comment by ngreenwald** [comment]
Got it. I've rolled back my version + docker image to the one I was using prior to the update. Is this something I should be trying to track down?

**Comment by willgraf** [comment]
This may be related to mounting your local code into the docker image, thus
overwriting the compiled C file.

Try not mounting the deepcell codebase and see if it works.

**Comment by ngreenwald** [comment]
Hi Will, that did the trick. 
I believe that since the new version switched from python3.5 to python3.6, the locally built version will need to be redone. 

Does that mean I need to run that setupy.py command myself locally before rebuilding the docker image?

**Comment by willgraf** [comment]
That's not quite what it means.  When you are mounting the source code with the line:

```
-v $PWD/deepcell:/usr/local/lib/python3.6/dist-packages/deepcell/ \
```
You are overwriting the inner `deepcell` module with your local copy.  This overwrites the `compute_overlap.c` file.  I find the best solution is to copy the compiled file from within a docker container to your local directory.  This way the expected file will be in the mounted directory.

1. Run a docker container without mounting
2. Copy the compute_overlap compiled file to your local directory: `docker cp container_name:/usr/local/lib/python3.6/dist-packages/deepcell/utils/compute_overlap.cpython-36m-x86_64-linux-gnu.so /path/to/deepcell-tf/deepcell/utils/compute_overlap.cpython-36m-x86_64-linux-gnu.so `
3. Kill the running container and remount the deepcell directory.

**Comment by willgraf** [comment]
Is this resolved?

**Comment by ngreenwald** [comment]
Yes, your solution above solved the issue! I updated the readme with the necessary instructions in #274. However, when I installed it locally on my laptop a couple weeks ago I didn't have to make the same change. Not sure if it's a linux-specific issue, or if the current version doesn't require the same steps.

**Comment by willgraf** [comment]
It might just be that you were not mounting the source code? I do not think it is a Linux specific issue.

I'm closing this issue as resolved. We can always link back to it if the compute_overlap flow is not working for users.

**Comment by Farnazmdi** [comment]
Hi!
I have a similar issue when trying to import `deepcell`. I am running it on a `virtualenv` on our (linux) server. Here is the error I got:
```
>>> import deepcell
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/farnazm/tHMM/venv/lib/python3.9/site-packages/deepcell/__init__.py", line 34, in <module>
    from deepcell import applications
  File "/home/farnazm/tHMM/venv/lib/python3.9/site-packages/deepcell/applications/__init__.py", line 32, in <module>
    from deepcell.applications.application import Application
  File "/home/farnazm/tHMM/venv/lib/python3.9/site-packages/deepcell/applications/application.py", line 37, in <module>
    from deepcell_toolbox.utils import resize, tile_image, untile_image
  File "/home/farnazm/tHMM/venv/lib/python3.9/site-packages/deepcell_toolbox/__init__.py", line 42, in <module>
    from deepcell_toolbox.compute_overlap import compute_overlap  # pylint: disable=E0401
  File "deepcell_toolbox/compute_overlap.pyx", line 1, in init deepcell_toolbox.compute_overlap
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject
```

Is this the same issue? I am not using a docker... Is it necessary?
Thank you.

**Comment by rafaelspeixoto** [comment]
@Farnazmdi Did you manage to fix this error? I am having the same issue.

**Comment by willgraf** [comment]
What version of Python are you using?

It might be caused by a `numpy` version incompatibility (as far as I can tell from this [StackOverflow post](https://play.hbomax.com/episode/urn:hbo:episode:GXtp2Cgdw8lhWhgEAAAAF))

**Comment by willgraf** [comment]
Regardless, I believe this is a separate issue than what @ngreenwald ran into. If it is still an issue for you, please create a new issue with all the system/platform details.

**Comment by Farnazmdi** [comment]
@rafaelspeixoto no, unfortuantely, I couldn't fix it and moved on.

**Comment by rafaelspeixoto** [comment]
Hey @Farnazmdi I was able to solve the problem. I first created a virtual environment and installed Deepcell, so it handled all of its dependencies and there were no conflicts. It worked fine in the virtual environment, but as I realized it was an issue regarding the versions of the package's dependencies, I updated all the libraries in my regular Python environment and downgraded the ones in conflict with Deepcell for the package to also work there.
